### ISSUE TITLE
Sage Attention is not installed

### ISSUE THREAD

**Issue by Keidan1993** [lllyasviel/FramePack#105]
Question I want to use Sage Attention but its saying I don't have it installed which I do. I tried installing both in C:\Users\alexl\AppData\Local\Programs\Python\Python310\Lib\site-packages and in the FramePack>system>python>lib>site packages, same with Triton saying it was successfully installed in the cmd. However when I start FramePack it still says "Sage Attention is not installed." I also installed CUDA toolkit 12.8 and Microsoft Visual studio and updated torch to 2.6. Does anyone know the problem?

I'm using python 310 so I installed triton_windows-3.2.0.post18-cp310-cp310-win_amd64.whl and sageattention-2.1.1+cu126torch2.6.0-cp310-cp310-win_amd64.whl. I'm using pip install in the cmd to install them in the locations I mentioned. I also updated pytorch, torchvision and the other one to 2.6. I did a restart, nothing. So what I'm I missing here?

**Comment by BitSoulLab** [comment]
You can use `pip list` to check whether you have install it.
I used Linux and manually install sageattention by `pip install sageattention`, and it worked.

**Comment by Keidan1993** [comment]
> You can use `pip list` to check whether you have install it. I used Linux and manually install sageattention by `pip install sageattention`, and it worked.

Thanks for the tip, I ran that command and it shows I have them all installed. I don't understand why FramePack is not recognizing it. If I do pip install sageattention its gonna install the v1 instead of v2. Unless thats the one it supports.

**Comment by Gothdir** [comment]
> Question I want to use Sage Attention but its saying I don't have it installed which I do. I tried installing both in C:\Users\alexl\AppData\Local\Programs\Python\Python310\Lib\site-packages and in the FramePack>system>python>lib>site packages, same with Triton saying it was successfully installed in the cmd. However when I start FramePack it still says "Sage Attention is not installed." I also installed CUDA toolkit 12.8 and Microsoft Visual studio and updated torch to 2.6. Does anyone know the problem?
> 
> I'm using python 310 so I installed triton_windows-3.2.0.post18-cp310-cp310-win_amd64.whl and sageattention-2.1.1+cu126torch2.6.0-cp310-cp310-win_amd64.whl. I'm using pip install in the cmd to install them in the locations I mentioned. I also updated pytorch, torchvision and the other one to 2.6. I did a restart, nothing. So what I'm I missing here?

Someone has made a good post about it installing packages https://github.com/lllyasviel/FramePack/issues/59#issuecomment-2815253240

**Comment by Keidan1993** [comment]
I have tried all of the suggestions but still is not detecting Sage Attention as installed.

**Comment by woct0rdho** [comment]
Hi @Keidan1993 , you need to make sure that you install them in the same Python environment that you run FramePack, see https://github.com/woct0rdho/triton-windows#2-python-environment

Do you use the one-click package and run by clicking `run.bat`? Then you should install everything in `framepack_cu126_torch26\system\python\`, and things in `C:\Users\alexl\AppData\Local\Programs\Python\Python310\` should not affect this.

Please run `C:\path\to\framepack_cu126_torch26\system\python\python.exe -m pip list` and paste all the results here

**Comment by wzwow** [comment]
![Image](https://github.com/user-attachments/assets/0285de50-937b-4455-9a90-eccbfabfd8d1)

flash attn not installed
sage attn not installed

what to do now? how do i resolve this?

`D:\AI\framepack\framepack_cu126_torch26\system\python>python.exe -m pip list
Package                   Version
------------------------- --------------------
absl-py                   2.1.0
accelerate                1.6.0
addict                    2.4.0
aenum                     3.1.15
aiofiles                  23.2.1
aiosignal                 1.3.1
albumentations            1.3.1
altair                    5.3.0
annotated-types           0.7.0
antlr4-python3-runtime    4.9.3
anyio                     3.7.1
async-timeout             4.0.3
attrs                     23.2.0
av                        12.1.0
basicsr                   1.4.2
blendmodes                2022
certifi                   2024.2.2
charset-normalizer        3.3.2
clean-fid                 0.1.35
click                     8.1.7
colorama                  0.4.6
coloredlogs               15.0.1
colorlog                  6.8.2
contourpy                 1.2.0
cycler                    0.12.1
Cython                    3.0.8
deprecation               2.1.0
depth_anything            2024.1.22.0
diffusers                 0.33.1
diskcache                 5.6.3
easydict                  1.11
einops                    0.4.1
embreex                   2.17.7.post4
exceptiongroup            1.2.0
facexlib                  0.3.0
fastapi                   0.115.12
ffmpy                     0.3.1
filelock                  3.13.1
filterpy                  1.4.5
fonttools                 4.47.2
frozenlist                1.4.1
fsspec                    2024.6.1
ftfy                      6.1.3
future                    0.18.3
gitdb                     4.0.11
gradio                    5.23.0
gradio_client             1.8.0
gradio_rangeslider        0.0.6
groovy                    0.1.2
grpcio                    1.60.1
h11                       0.12.0
handrefinerportable       2024.2.12.0
httpcore                  0.15.0
httpx                     0.24.1
huggingface-hub           0.30.2
humanfriendly             10.0
idna                      3.6
imageio                   2.33.1
importlib-metadata        7.0.1
inflection                0.5.1
insightface               0.7.3
Jinja2                    3.1.4
joblib                    1.3.2
jsonmerge                 1.8.0
jsonschema                4.21.1
jsonschema-specifications 2023.12.1
kiwisolver                1.4.5
kornia                    0.6.7
lark                      1.1.2
lazy_loader               0.3
lightning-utilities       0.10.1
llvmlite                  0.42.0
lmdb                      1.4.1
mapbox-earcut             1.0.1
Markdown                  3.5.2
markdown-it-py            3.0.0
MarkupSafe                2.1.5
matplotlib                3.8.2
mdurl                     0.1.2
mpmath                    1.3.0
multidict                 6.0.5
networkx                  3.3
numba                     0.59.0
numpy                     1.26.2
omegaconf                 2.2.3
onnx                      1.15.0
onnxruntime               1.17.0
opencv-contrib-python     4.9.0.80
opencv-python             4.10.0.84
opencv-python-headless    4.9.0.80
orjson                    3.9.13
packaging                 23.2
pandas                    2.2.0
piexif                    1.1.3
pillow                    11.1.0
pillow-avif-plugin        1.4.3
pip                       25.0.1
platformdirs              4.2.0
prettytable               3.9.0
protobuf                  3.20.0
psutil                    5.9.5
pycollada                 0.8
pydantic                  2.8.2
pydantic_core             2.20.1
pydub                     0.25.1
Pygments                  2.18.0
pyparsing                 3.1.1
pyreadline3               3.4.1
python-dateutil           2.8.2
python-multipart          0.0.20
pytz                      2024.1
PyWavelets                1.5.0
PyYAML                    6.0.1
qudida                    0.0.4
referencing               0.33.0
regex                     2023.12.25
requests                  2.31.0
resize-right              0.0.2
rich                      13.7.1
rpds-py                   0.17.1
Rtree                     1.2.0
ruff                      0.11.6
safehttpx                 0.1.6
safetensors               0.5.3
scikit-image              0.21.0
scikit-learn              1.4.0
scipy                     1.12.0
semantic-version          2.10.0
sentencepiece             0.2.0
setuptools                69.5.1
shapely                   2.0.2
shellingham               1.5.4
six                       1.16.0
smmap                     5.0.1
sniffio                   1.3.0
spandrel                  0.3.4
spandrel_extra_arches     0.1.1
starlette                 0.46.2
svg.path                  6.3
sympy                     1.13.1
tb-nightly                2.16.0a20240204
tensorboard-data-server   0.7.2
tf_keras-nightly          2.16.0.dev2024020410
threadpoolctl             3.2.0
tifffile                  2024.1.30
tokenizers                0.20.3
tomli                     2.0.1
tomlkit                   0.12.0
toolz                     0.12.1
torch                     2.6.0+cu126
torchaudio                2.6.0+cu126
torchdiffeq               0.2.3
torchmetrics              1.3.0.post0
torchsde                  0.2.6
torchvision               0.21.0+cu126
tqdm                      4.67.1
trampoline                0.1.2
transformers              4.46.2
trimesh                   4.1.3
typer                     0.12.3
typing_extensions         4.12.2
tzdata                    2023.4
urllib3                   2.2.0
uvicorn                   0.27.0.post1
vhacdx                    0.0.5
wcwidth                   0.2.13
websockets                11.0.3
Werkzeug                  3.0.1
wheel                     0.42.0
yapf                      0.40.2
yarl                      1.9.4
zipp                      3.17.0`

**Comment by woct0rdho** [comment]
@wzwow For the one-click package, you can do the following:

Download the two folders 'include' and 'libs' here https://github.com/woct0rdho/triton-windows/releases/download/v3.0.0-windows.post1/python_3.10.11_include_libs.zip and copy them to `framepack_cu126_torch26\system\python\` (This is only needed for the one-click package, not the usual Python installation)

Then run:
```
D:\AI\framepack\framepack_cu126_torch26\system\python\python.exe -m pip install -U 'triton-windows<3.3'
D:\AI\framepack\framepack_cu126_torch26\system\python\python.exe -m pip install -U https://github.com/woct0rdho/SageAttention/releases/download/v2.1.1-windows/sageattention-2.1.1+cu126torch2.6.0-cp310-cp310-win_amd64.whl
```

Then run FramePack and you should see 'Sage Attn is installed!'

**Comment by Keidan1993** [comment]
> Hi [@Keidan1993](https://github.com/Keidan1993) , you need to make sure that you install them in the same Python environment that you run FramePack, see https://github.com/woct0rdho/triton-windows#2-python-environment
> 
> Do you use the one-click package and run by clicking `run.bat`? Then you should install everything in `framepack_cu126_torch26\system\python\`, and things in `C:\Users\alexl\AppData\Local\Programs\Python\Python310\` should not affect this.
> 
> Please run `C:\path\to\framepack_cu126_torch26\system\python\python.exe -m pip list` and paste all the results here

Package                   Version
------------------------- ---------------------
absl-py                   2.1.0
accelerate                1.6.0
addict                    2.4.0
aenum                     3.1.15
aiofiles                  23.2.1
aiosignal                 1.3.1
albumentations            1.3.1
altair                    5.3.0
annotated-types           0.7.0
antlr4-python3-runtime    4.9.3
anyio                     3.7.1
async-timeout             4.0.3
attrs                     23.2.0
av                        12.1.0
basicsr                   1.4.2
blendmodes                2022
certifi                   2024.2.2
charset-normalizer        3.3.2
clean-fid                 0.1.35
click                     8.1.7
colorama                  0.4.6
coloredlogs               15.0.1
colorlog                  6.8.2
contourpy                 1.2.0
cycler                    0.12.1
Cython                    3.0.8
deprecation               2.1.0
depth_anything            2024.1.22.0
diffusers                 0.33.1
diskcache                 5.6.3
easydict                  1.11
einops                    0.4.1
embreex                   2.17.7.post4
exceptiongroup            1.2.0
facexlib                  0.3.0
fastapi                   0.115.12
ffmpy                     0.3.1
filelock                  3.13.1
filterpy                  1.4.5
flash_attn                2.7.4
fonttools                 4.47.2
frozenlist                1.4.1
fsspec                    2024.6.1
ftfy                      6.1.3
future                    0.18.3
gitdb                     4.0.11
gradio                    5.23.0
gradio_client             1.8.0
gradio_rangeslider        0.0.6
groovy                    0.1.2
grpcio                    1.60.1
h11                       0.12.0
handrefinerportable       2024.2.12.0
httpcore                  0.15.0
httpx                     0.24.1
huggingface-hub           0.30.2
humanfriendly             10.0
idna                      3.6
imageio                   2.33.1
importlib-metadata        7.0.1
inflection                0.5.1
inquirerpy                0.3.4
insightface               0.7.3
Jinja2                    3.1.4
joblib                    1.3.2
jsonmerge                 1.8.0
jsonschema                4.21.1
jsonschema-specifications 2023.12.1
kiwisolver                1.4.5
kornia                    0.6.7
lark                      1.1.2
lazy_loader               0.3
lightning-utilities       0.10.1
llvmlite                  0.42.0
lmdb                      1.4.1
mapbox-earcut             1.0.1
Markdown                  3.5.2
markdown-it-py            3.0.0
MarkupSafe                2.1.5
matplotlib                3.8.2
mdurl                     0.1.2
mpmath                    1.3.0
multidict                 6.0.5
networkx                  3.3
numba                     0.59.0
numpy                     1.26.2
omegaconf                 2.2.3
onnx                      1.15.0
onnxruntime               1.17.0
opencv-contrib-python     4.9.0.80
opencv-python             4.10.0.84
opencv-python-headless    4.9.0.80
orjson                    3.9.13
packaging                 23.2
pandas                    2.2.0
pfzy                      0.3.4
piexif                    1.1.3
pillow                    11.1.0
pillow-avif-plugin        1.4.3
pip                       25.0.1
platformdirs              4.2.0
prettytable               3.9.0
prompt_toolkit            3.0.51
protobuf                  3.20.0
psutil                    5.9.5
pycollada                 0.8
pydantic                  2.8.2
pydantic_core             2.20.1
pydub                     0.25.1
Pygments                  2.18.0
pyparsing                 3.1.1
pyreadline3               3.4.1
python-dateutil           2.8.2
python-multipart          0.0.20
pytz                      2024.1
PyWavelets                1.5.0
PyYAML                    6.0.1
qudida                    0.0.4
referencing               0.33.0
regex                     2023.12.25
requests                  2.31.0
resize-right              0.0.2
rich                      13.7.1
rpds-py                   0.17.1
Rtree                     1.2.0
ruff                      0.11.6
safehttpx                 0.1.6
safetensors               0.5.3
sageattention             2.1.1+cu126torch2.6.0
scikit-image              0.21.0
scikit-learn              1.4.0
scipy                     1.12.0
semantic-version          2.10.0
sentencepiece             0.2.0
setuptools                69.5.1
shapely                   2.0.2
shellingham               1.5.4
six                       1.16.0
smmap                     5.0.1
sniffio                   1.3.0
spandrel                  0.3.4
spandrel_extra_arches     0.1.1
starlette                 0.46.2
svg.path                  6.3
sympy                     1.13.1
tb-nightly                2.16.0a20240204
tensorboard-data-server   0.7.2
tf_keras-nightly          2.16.0.dev2024020410
threadpoolctl             3.2.0
tifffile                  2024.1.30
tokenizers                0.20.3
tomli                     2.0.1
tomlkit                   0.12.0
toolz                     0.12.1
torch                     2.6.0+cu126
torchaudio                2.6.0+cu126
torchdiffeq               0.2.3
torchmetrics              1.3.0.post0
torchsde                  0.2.6
torchvision               0.21.0+cu126
tqdm                      4.67.1
trampoline                0.1.2
transformers              4.46.2
trimesh                   4.1.3
triton-windows            3.2.0.post18
typer                     0.12.3
typing_extensions         4.12.2
tzdata                    2023.4
urllib3                   2.2.0
uvicorn                   0.27.0.post1
vhacdx                    0.0.5
wcwidth                   0.2.13
websockets                11.0.3
Werkzeug                  3.0.1
wheel                     0.42.0
xformers                  0.0.29.post3
yapf                      0.40.2
yarl                      1.9.4
zipp                      3.17.0

**Comment by woct0rdho** [comment]
@Keidan1993  Are these the results of `C:\path\to\framepack_cu126_torch26\system\python\python.exe -m pip list`? Please also show how you run this command, as in wzwow's post.

Also, do you run FramePack by clicking `run.bat`?

**Comment by Keidan1993** [comment]
> [@Keidan1993](https://github.com/Keidan1993) Are these the results of `C:\path\to\framepack_cu126_torch26\system\python\python.exe -m pip list`? Please also show how you run this command, as in wzwow's post.
> 
> Also, do you run FramePack by clicking `run.bat`?

I do run it by clicking run.bat, also if I run that command exactly like that this is what I get:

![Image](https://github.com/user-attachments/assets/616937f1-b95b-4bb7-a3fe-92184deab81c)

**Comment by woct0rdho** [comment]
Ok you need to change `C:\path\to\framepack_cu126_torch26` to the actual path of your `framepack_cu126_torch26` folder

**Comment by Keidan1993** [comment]
> Ok you need to change `C:\path\to\framepack_cu126_torch26` to the actual path of your `framepack_cu126_torch26` folder

I did do that, and I got what I posted above.

**Comment by woct0rdho** [comment]
Ok, it seems you're running `pip` in the correct way. Let's check a few other things:

Run `C:\Users\alexl\Downloads\framepack_cu126_torch26\system\python\python.exe` to start Python, then run the following:
```python
import triton
import sageattention
print(dir(sageattention.core))
print(sageattention.core.SM80_ENABLED)
print(sageattention.core.SM89_ENABLED)
print(sageattention.core.SM90_ENABLED)
```
and show all the results.

In my case, it shows the following:
![Image](https://github.com/user-attachments/assets/d25e5ff6-4f35-4983-94cb-4e72bcac580c)

**Comment by Keidan1993** [comment]
> Ok, it seems you're running `pip` in the correct way. Let's check a few other things:
> 
> Run `C:\Users\alexl\Downloads\framepack_cu126_torch26\system\python\python.exe` to start Python, then run the following:
> 
> import triton
> import sageattention
> print(dir(sageattention.core))
> print(sageattention.core.SM80_ENABLED)
> print(sageattention.core.SM89_ENABLED)
> print(sageattention.core.SM90_ENABLED)
> and show all the results.
> 
> In my case, it shows the following: ![Image](https://github.com/user-attachments/assets/d25e5ff6-4f35-4983-94cb-4e72bcac580c)

This is what I got:
